lib/pbench/server/database/models/datasets.py

```diff
@@ -895,6 +899,8 @@ def is_key_path(key: str, valid: List[str]) -> bool:
             True if the path is valid, or False
         """
         k = key.lower()
+        if not re.match(Metadata._valid_key_charset, k):
```

Comment:
I recommend omitting the anchors from `_valid_key_charset` and using `fullmatch()` instead of `match()` here, as it is more expressive; doing so will save us from having to discuss whether and how `\Z` is better than `$`....